lib/publicReportingAPI.js

```diff
@@ -0,0 +1,22 @@
+const { EVENTS } = require('./constants');
+const { publishEvent } = require('./utils');
+
+class PublicReportingAPI {
+    static addDescription(text) {
```

Comment:
These methods can be part of the `RPClient` class

Reply:
I'd preffer don't mix client and api class.
- because RPClient already overwhelmed
- for agent users we explicitly split what is private what is public.

Reply:
What is your suggestion?

Reply:
The idea was to create the API for agents to implement and expose implemented methods to users.

Reply:
>What is your suggestion?

I am wondering about RPClient size and content. It is 600+ LoC already. We have separate rest client, analytics and some helpers - it is good.
I am not sure it is good idea to add more things to it.
Such api methods can be easily moved to separate class and have more meaningful name rather than RPClient - because it is not client to rp :) it is some api for agents.

btw it is just my thoughts. I dont really know how you are gonna use\implement these APIs in agents.